src/common/src/catalog/column.rs

```diff
@@ -211,6 +211,38 @@ impl From<&ColumnDesc> for ProstColumnDesc {
     }
 }
 
+impl From<&DataType> for ColumnDesc {
+    fn from(dtype: &DataType) -> Self {
+        let fields_desc = if let DataType::Struct(fields) = dtype {
+            let mut field_descs = fields
+                .fields
+                .iter()
+                .map(ColumnDesc::from)
+                .collect::<Vec<ColumnDesc>>();
+            if !fields.field_names.is_empty() {
+                field_descs = field_descs
+                    .into_iter()
+                    .zip_eq(fields.field_names.clone())
+                    .map(|(mut desc, name)| {
+                        desc.name = name;
+                        desc
+                    })
+                    .collect::<Vec<ColumnDesc>>();
+            }
+            field_descs
+        } else {
+            Vec::with_capacity(0)
```

Comment:
May simply use `Vec::new()` or `vec![]`. For empty string, use `String::new()`, `Default::default()`, `"".to_owned()`, or `"".to_string()`.

Reply:
Thanks, I mistakenly thought that Vec::new would cause memory allocation.

Reply:
It's a `const fn` so don't worry about it. 🥰